parcels/field.py

```diff
@@ -664,13 +673,24 @@ def _time_index(self, time):
         if time_index.all():
             # If given time > last known field time, use
             # the last field frame without interpolation
-            return len(self.grid.time) - 1
+            ti = len(self.grid.time) - 1
         elif np.logical_not(time_index).all():
             # If given time < any time in the field, use
             # the first field frame without interpolation
-            return 0
+            ti = 0
+        else:
+            ti = time_index.argmin() - 1 if time_index.any() else 0
+        if self.grid.tdim == 1:
+            tau = 0
+        elif ti == len(self.grid.time) - 1:
+            tau = 1
         else:
-            return time_index.argmin() - 1 if time_index.any() else 0
+            tau = (
+                (time - self.grid.time[ti]) / (self.grid.time[ti + 1] - self.grid.time[ti])
+                if self.grid.time[ti] != self.grid.time[ti + 1]
+                else 0
+            )
+        return tau, ti
```

Comment:
Could we either:
- Update the docstring and function name for `_time_index` reflecting it also returns tau, or
- Split the tau calculation into a `_get_tau(grid, ti)` function

?

(slight preference for the latter)

Reply:
I prefer the first option, as the other (spatial) search_indices functions also return both the index and the relative position; so a `search_time_index` should return the same `(int, float)` type, I feel

Reply:
Ok, sounds good